YoutubeExplode/YoutubeClient.cs

```diff
@@ -186,6 +186,44 @@ public YoutubeClient()
         }
 
         /// <summary>
+        /// Gets all videos in a given playlist.
+        /// This method will does one GET request per every 20 videos in the playlist.
+        /// 
+        /// This also works for special playlists like the Watch Later playlist at https://www.youtube.com/playlist?list=WL,
+        /// but the _requestServices has to mimic a logged in user.
+        /// Same for other non public playlists.
+        /// </summary>
+        public async Task<List<string>> GetVideoIdsInPlaylistAsync(string playlistId)
+        {
+            var videoIds = new List<string>();
+
+            //We use a youtube api that is also used by the mobile version of the youbute page (m.youtube.com).
+            //For this to work we need to add &ajax=1&tsp=1&app=m to our playlist url
+            var baseUrl = $"https://m.youtube.com/playlist?list={playlistId}&ajax=1&tsp=1&app=m";
+            var nextUrl = baseUrl;
+            while (nextUrl != null)
+            {
+                var responseContent = await _requestService.GetStringAsync(nextUrl).ConfigureAwait(false);
+                nextUrl = null;
+
+                var videoIdMatches = Regex.Matches(responseContent, @"""video_id"": ""([0-9a-zA-Z_\-]+)""");
+                var currentVideoIds = videoIdMatches.Cast<Match>()
+                                             .Where(m => m.Groups[1].Success)
+                                             .Select(m => m.Groups[1].Value)
+                                             .Distinct(); //VideoIds occur multiple times in the json data make sure we have not dupplicates
```

Comment:
Wouldn't it make more sense to `.Distinct` the final `videoIds` list instead? It also covers edge cases when the same ids appear on different pages for whatever reason.